modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/H2ResultSetIterator.java

```diff
@@ -180,6 +184,9 @@ private boolean fetchPage() throws IgniteCheckedException {
                     page.add(row);
                 }
                 catch (SQLException e) {
+                    if (X.hasCause(e, QueryMemoryTracker.TrackerWasClosedException.class) && canceled)
```

Comment:
Is it safe to skip Iterator close() call?

Reply:
No itn't. I've move this check after close() invoking.